src/lib/comms/include/sol-mqtt.h

```diff
@@ -341,11 +346,10 @@ struct sol_mqtt_config {
  * @param host The host address of the MQTT broker
  * @param port The host port to connect to
  * @param config Configuration and callbacks
- * @param data User data provided to the callbacks
  *
  * @return New mqtt object on sucess, NULL otherwise
  */
-struct sol_mqtt *sol_mqtt_connect(const char *host, uint16_t port, const struct sol_mqtt_config *config, const void *data);
+struct sol_mqtt *sol_mqtt_connect(const char *host, uint16_t port, const struct sol_mqtt_config *config);
```

Comment:
@barbieri this is the connect function, i guess we can keep the address out of the config